src/CompletionProvider.php

```diff
@@ -280,114 +287,278 @@ public function provideCompletion(PhpDocument $doc, Position $pos, CompletionCon
+            $currentNamespace = $namespaceNode === null ? '' : $namespaceNode->name->getText();
 
-            /** Whether the prefix is fully qualified (begins with a backslash) */
-            $isFullyQualified = $nameNode instanceof Node\QualifiedName && $nameNode->isFullyQualifiedName();
+            /** @var bool Whether the prefix is qualified (contains at least one backslash) */
+            $isFullyQualified = false;
 
-            /** The closest NamespaceDefinition Node */
-            $namespaceNode = $node->getNamespaceDefinition();
+            /** @var bool Whether the prefix is qualified (contains at least one backslash) */
+            $isQualified = false;
 
-            /** @var string The name of the namespace */
-            $namespacedPrefix = null;
-            if ($namespaceNode) {
-                $namespacedPrefix = (string)PhpParser\ResolvedName::buildName($namespaceNode->name->nameParts, $node->getFileContents()) . '\\' . $prefix;
-                $namespacedPrefixLen = strlen($namespacedPrefix);
+            if ($nameNode instanceof Node\QualifiedName) {
+                $isFullyQualified = $nameNode->isFullyQualifiedName();
+                $isQualified = $nameNode->isQualifiedName();
             }
 
-            // Get the namespace use statements
-            // TODO: use function statements, use const statements
+            /** @var bool Whether we are in a new expression */
+            $isCreation = isset($creation);
+
+            /** @var array Import (use) tables */
+            $importTables = $node->getImportTablesForCurrentScope();
+
+            if ($isFullyQualified) {
+                // \Prefix\Goes\Here| - Only return completions from the root namespace.
+                /** @var $items \Generator|CompletionItem[] Generator yielding CompletionItems indexed by their FQN */
+                $items = $this->getCompletionsForFqnPrefix($prefix, $isCreation, false);
+            } else if ($isQualified) {
+                // Prefix\Goes\Here|
+                $items = $this->getPartiallyQualifiedCompletions(
+                    $prefix,
+                    $currentNamespace,
+                    $importTables,
+                    $isCreation
+                );
+            } else {
+                // PrefixGoesHere|
+                $items = $this->getUnqualifiedCompletions($prefix, $currentNamespace, $importTables, $isCreation);
+            }
 
-            /** @var string[] $aliases A map from local alias to fully qualified name */
-            list($aliases,,) = $node->getImportTablesForCurrentScope();
+            $list->items = array_values(iterator_to_array($items));
+            foreach ($list->items as $item) {
+                // Remove ()
+                if (is_string($item->insertText) && substr($item->insertText, strlen($item->insertText) - 2) === '()') {
+                    $item->insertText = substr($item->insertText, 0, strlen($item->insertText) - 2);
```

Comment:
no need to use `strlen` here(this line and the line before).

Use `substr($item->insertText, -2);` instead.